coffea/processor/servicex/executor.py

```diff
@@ -0,0 +1,197 @@
+            # in uproot4 that means `file://` isn't parsed correctly on windows.
+            # TODO: Remove this hack when `uproot4` has been updated.
+            p = urlparse(file_url)
+            if p.scheme == "file":
+                file_url = url2pathname(unquote(p.path))
+
+            # Determine the tree name if we've not gotten it already
+            if tree_name is None:
+                with uproot.open(file_url) as sample:
+                    tree_name = sample.keys()[0]
+
+            # Invoke the implementation's task launcher
+            data_result = self.run_async_analysis(
+                file_url=file_url,
+                tree_name=tree_name,
+                accumulator=accumulator,
+                process_func=process_func,
+            )
+
+            # Pass this down to the next item in the stream.
+            yield data_result
+
+    async def accumulate(
+        self, analysis: Analysis, hist_stream: AsyncGenerator[Accumulator, None]
+    ) -> AsyncGenerator[Accumulator, None]:
+        """
+        Stream processor to accumulate histograms from each process task and yield the
+        up-to-date accumulated histograms
+        :param analysis: Analysis
+            The analysis instance being accumulated
+        :param hist_stream:  AsyncGenerator[Accumulator]
+            Stream of histograms from each process task
+        :return:
+        """
+        # There is an accumulate pattern in the aiostream lib
+        output = analysis.accumulator.identity()
+        async with hist_stream.stream() as streamer:
+            async for results in streamer:
+                print("$$$ ", results)
+                output.add(results)
+                yield output
+
+
+def run_coffea_processor(
+    events_url: str, tree_name: str, accumulator, proc, explicit_func_pickle=False
+):
+    """
+    Process a single file from a tree via a coffea processor on the remote node
+    :param events_url:
+        a URL to a ROOT file that uproot4 can open
+    :param tree_name:
+        The tree in the ROOT file to use for our data
+    :param accumulator:
+        Accumulator to store the results
+    :param proc:
+        Analysis function to execute. Must have signature
+    :param explicit_func_pickle: bool
+        Do we need to use dill to explicitly pickle the process function, or can we
+        rely on the remote execution framework to handle it correctly?
+    :return:
+        Populated accumulator
+    """
+    # Since we execute remotely, explicitly include everything we need.
+    import awkward as ak
+    from coffea.nanoevents import NanoEventsFactory
+    from coffea.processor.servicex.schema import auto_schema
+
+    # This in is amazingly important - the invar mass will fail silently without it.
+    # And must be done in here as this function is shipped off to the funcx processor
+    # on a remote machine/remote python environment.
+    from coffea.nanoevents.methods import candidate
```

Comment:
This can/should be removed once my comment on the schema is implemented. Though I'm surprised its "silent"--awkward has since at least a little while produced an error on addition of record arrays without an explicit overloadd.